src/discussions/in-context-topics/TopicsView.test.jsx

```diff
@@ -0,0 +1,227 @@
+            </Route>
+            <Route
+              render={({ location }) => {
+                lastLocation = location;
+                return null;
+              }}
+            />
+          </MemoryRouter>
+        </DiscussionContext.Provider>
+      </AppProvider>
+    </IntlProvider>,
+  );
+  container = component.container;
+  return container;
+}
+
+describe('Legacy Topics View', () => {
+  let nonCoursewareTopics;
+  let coursewareTopics;
+  beforeEach(() => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: true,
+        roles: [],
+      },
+    });
+
+    store = initializeStore({
+      config: { enableInContext: true, provider: 'openedx', hasModerationPrivileges: true },
+    });
+    Factory.resetAll();
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+    lastLocation = undefined;
+  });
+
+  async function setupMockResponse() {
+    axiosMock.onGet(`${topicsApiUrl}${courseId}`)
+      .reply(200, (Factory.buildList('topic', 1, null, {
+        topicPrefix: 'noncourseware-topic',
+        enabledInContext: true,
+        topicNamePrefix: 'general-topic',
+        usageKey: '',
+        courseware: false,
+        discussionCount: 1,
+        questionCount: 1,
+      }).concat(Factory.buildList('section', 2, null, { topicPrefix: 'courseware' })))
+        .concat(Factory.buildList('archived-topics', 2, null)));
+    await executeThunk(fetchCourseTopicsV3(courseId), store.dispatch, store.getState);
+
+    const state = store.getState();
+    nonCoursewareTopics = selectNonCoursewareTopics(state);
+    coursewareTopics = selectCoursewareTopics(state);
+  }
+
+  it('A non-courseware topic should be clickable and should have a title', async () => {
+    await setupMockResponse();
+    renderComponent();
+
+    nonCoursewareTopics.forEach(async topic => {
+      const noncoursewareTopic = await screen.findByText(topic.name);
+
+      await act(async () => fireEvent.click(noncoursewareTopic));
+      await waitFor(async () => {
+        expect(screen.queryByText(topic.name)).toBeInTheDocument();
+        expect(lastLocation.pathname.endsWith(`/topics/${topic.id}`)).toBeTruthy();
+      });
+    });
+  });
+
+  it('A non-courseware topic should be on the top of the list', async () => {
+    await setupMockResponse();
+    renderComponent();
+    const topicsList = await screen.getByRole('list');
+    const { firstChild } = topicsList;
+    expect(within(firstChild).queryByText(nonCoursewareTopics[0].name)).toBeInTheDocument();
+    expect(topicsList.children[1]).toBe(topicsList.querySelector('.divider'));
+  });
+
+  it('A non-Courseware topic should have 3 stats and should be hoverable', async () => {
+    await setupMockResponse();
+    renderComponent();
+    const topicsList = await screen.getByRole('list');
+    const { firstChild } = topicsList;
+    const statsList = await firstChild.getElementsByClassName('pgn__icon');
```

Comment:
const topic = await container.querySelector('.discussion-topic'); // It will give you first nonCoursewareTopic
newLine
const statsList = await topic.querySelectorAll('.icon-size'); // pgn__icon is a paragon class it is not a good idea to use it instead use our own define class.